share/python/procedures/findif_response_utils/tensor_helper.py

```diff
@@ -0,0 +1,118 @@
+#
+# @BEGIN LICENSE
+#
+# Psi4: an open-source quantum chemistry software package
+#
+# Copyright (c) 2007-2016 The Psi4 Developers.
+#
+# The copyrights for code used from other parties are included in
+# the corresponding files.
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, write to the Free Software Foundation, Inc.,
+# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.
+#
+# @END LICENSE
+#
+
+"""
+Module of helper functions for distributed ccresponse computations.
+
+Defines functions for retrieving tensors computed at displaced geometries.
+"""
+from __future__ import absolute_import
+from __future__ import print_function
+import collections
+import shelve
+import copy
+import os
+import psi4
+import p4util
+from p4const import *
+
+
+def synthesize_displaced_tensor(db, signature, row_dim):
+    """
+        Gathers a list of tensors, one at each displaced geometry.
+
+    db: (database) the database object for this property calculation
+    signature: (string) The string that notifies the matrix reader that the
+        targeted tensor data begins.
+    row_dim: the expected number of rows that this value should be printed
+        across in the file
+
+    Returns a 2d list result[i][j]:
+        i: indexes displacements
+        j: indexes elements of the flattened tensor at some displacement
+    Throws: none
+    """
+    result = []
+    for job in db['job_status']:
+        with open('{}/output.dat'.format(job)) as outfile:
+            result.append(grab_psi4_matrix(outfile, signature, row_dim))
+
+    return result
+
+    # END sythesize_displaced_tensor()
+
+
+def grab_psi4_matrix(outfile, matrix_name, row_tot):
```

Comment:
I would point out that this is not for general Psi4 matrices, but very specifically for (n x 3) geometry-like matrices and the return is vector-like no matrix-like. Perhaps `parse_psi4_geometry_matrix` for the name.